### PR TITLE
Initial update of JMX's Kafka metrics.

### PR DESCRIPTION
**Pull request by carlosalberto**
PR with the upcoming changes from https://github.com/open-telemetry/opentelemetry-specification/pull/2485 - observe not all metrics are included, namely:

* Metrics which could use Summary points instead (changes have been postponed)
* Consumer metrics (follow up required)
* ALL metrics now have a `messaging` prefix, in order to comply with the tracing semantic conventions.

Please review @dehaansa @rmfitzpatrick

### CODE REVIEW

**Comment by carlosalberto** [comment]
@rmfitzpatrick @dehaansa Please review, as the actual Specification has been updated.

**Comment by laurit** [comment]
> I'm getting failed checks because of this, which is not related to the changes themselves. Any idea?
> 
> ```shell
> Task :static-instrumenter:agent-extension:compileJava
> /home/runner/work/opentelemetry-java-contrib/opentelemetry-java-contrib/static-instrumenter/agent-extension/src/main/java/io/opentelemetry/contrib/staticinstrumenter/extension/AdditionalClassesInjectorListenerInstaller.java:23: warning: [deprecation] beforeAgent(Config,AutoConfiguredOpenTelemetrySdk) in AgentListener has been deprecated
>   public void beforeAgent(
>               ^
> error: warnings found and -Werror specified
> 1 error
> 1 warning
> ```

@carlosalberto hopefully fixed, try rebasing

**Comment by carlosalberto** [comment]
@laurit Awesome, thanks! Done.

**Review by trask — APPROVED** [review]
@carlosalberto thanks for driving these into the spec!

**Comment by rmfitzpatrick** [comment]
Thanks for driving the spec updates and these changes, which look great. Do you mind updating [the existing readmes](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/jmx-metrics/docs/target-systems), maybe linking to the spec for the producer and general metrics to reduce c/p?

File: jmx-metrics/src/main/resources/target-systems/kafka.groovy [hunk 1]
```diff
@@ -161,29 +161,29 @@ otel.instrument(isrOperations,
 
 
 def maxLag = otel.mbean("kafka.server:type=ReplicaFetcherManager,name=MaxLag,clientId=Replica")
-otel.instrument(maxLag, "kafka.max.lag", "max lag in messages between follower and leader replicas",
-        "{messages}", "Value", otel.&longValueCallback)
+otel.instrument(maxLag, "messaging.kafka.max.lag", "max lag in messages between follower and leader replicas",
```
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply jmx-metrics/src/main/resources/target-systems/kafka.groovy:164]: In the connection pool PR I treated the description as a part of the instrument specification, maybe a little bit similar to the unit.

[the remark this answers is not in the document]
Comment (carlosalberto) [reply jmx-metrics/src/main/resources/target-systems/kafka.groovy]: Good call @trask, just did a final pass to compare/fix with the actual values in the Spec.

**Comment by carlosalberto** [comment]
@rmfitzpatrick Are they auto generated? I'm guessing that's a NO? Updating them manually doesn't sound very fun, etc.

**Comment by rmfitzpatrick** [comment]
> @rmfitzpatrick Are they auto generated? I'm guessing that's a NO? Updating them manually doesn't sound very fun, etc.

Right, we don't have a metadata scheme/docs generator for the project atm. That would be a nice enhancement.

**Comment by carlosalberto** [comment]
I saw 1.14 was just released - so it means one more month to go with the out of date metrics, right?

(And yes, the docs should be autogenerated - it's enough pain to update them in the code and tests, and then go through that in the docs as well).

**Comment by trask** [comment]
@carlosalberto also are you still working on this? thx

**Pull request closed without merging**